lib/client-utils.ts

```diff
@@ -0,0 +1,156 @@
+import { HabitT } from './types-and-constants';
+
+/**
+ * @param date is the refrance date from which the days of week are going to get created
+ * @return dates and formetedays
+ */
+export function generateWeekDays(date: Date): Date[] {
+    const dayOfWeek = date.getDay(); // 0 = Sunday, 1 = Monday, ..., 6 = Saturday
+    const diff = date.getDate() - dayOfWeek + (dayOfWeek === 0 ? -6 : 1); // Adjust to Monday
+
+    const monday = new Date(date.setDate(diff));
+    const sunday = new Date(date.setDate(diff + 6));
+
+    const weekDates: Date[] = [];
+    for (let i = 0; i <= 6; i++) {
+        const currentDate = new Date(monday);
+        currentDate.setDate(currentDate.getDate() + i);
+        weekDates.push(currentDate);
+    }
+
+    return weekDates;
+}
+
+export function isWithinLast7Days(dateToCheck: Date): boolean {
+    const currentDate = new Date();
+    const sevenDaysAgo = new Date();
+    sevenDaysAgo.setDate(currentDate.getDate() - 6);
+
+    // Set hours, minutes, seconds, and milliseconds to 0 for both dates
+    currentDate.setHours(0, 0, 0, 0);
+    sevenDaysAgo.setHours(0, 0, 0, 0);
+    dateToCheck.setHours(0, 0, 0, 0);
+
+    return !(dateToCheck >= sevenDaysAgo && dateToCheck <= currentDate);
+}
+
+export const checkedDatesMap = (habits: HabitT[]) => {
+    // console.log('checkedDatesMap called');
+
+    return habits.reduce(
+        (mapedDates, habit) => {
+            habit.checkmarks.forEach((checkmark) => {
+                const date = new Date(checkmark.date);
+                const dateString = date.toDateString();
+                if (!mapedDates[dateString]) {
+                    mapedDates[dateString] = [];
+                }
+
+                mapedDates[dateString].push(habit.id);
+            });
+
+            return mapedDates;
+        },
+        {} as { [key: string]: number[] }
+    );
+};
+
+type CheckedDateItem = {
+    id: number;
+    date: string;
+    habitId: number;
+    checkmark: boolean;
+};
+
+/**
+ * If the string is bigger than the max length, the string is cut and the threee dots are added at the end
+ * If the string is not bigger than the maxt length, then just returns the string
+ */
+export const setMaxStringLength = (text: string, maxLength: number): string => {
```

Comment:
utils.ts is a server files which can makes its functions unuseable to client side components

Reply:
Since utils.ts is designated with "import 'server-only';" indicating it's inaccessible for client-side components, we've separated out a dedicated utils file for client-side use. To avoid duplication and maintain consistency, we've removed the setMaxStringLength function from utils.ts. This ensures that the functionality is appropriately situated for both server and client needs.